### PR TITLE
Create a basic list of tools and configuration for developers

### PR DESCRIPTION
**Pull request by alexchapar**
Draft for some tooling that can be useful for devevelopers. It's essentially intended for developers on Windows ( but not limited to them 😉)

It includes configuration for Git, some useful registry, scripts to manage services, scripts and helper to work with Scoop.sh and also managing the extensions of VS Code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,21 @@
...
+
+> It is still a `WORK IN PROGRESS`
+
+## Getting Started
+
+For the moment, to use it, just clone the repo... From there, you should be "Good to go".
+
+Then execute the scripts. Each folder should contains a readme that explains how to use it.
+
+By the way, We know, it's not elegant but we do trust you will contribute with some PR to improve it :P
```
Comment (ghost) [inline README.md]: Trust your tools! :) Maybe update the sentence to something more positive? Something like "We're always improving and your contributions are welcome!"?

File: git/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Configuration for Git
+
+[[_TOC_]]
+
+## Overview
+
+This folder contains configuration for git and also the shell integration of the git-bash (using the `reg` files)
+
+These files are configured with the assumption that you are using [SCOOP](https://scoop.sh), as we do ;)
```
Comment (ghost) [inline git/README.md]: "as we do" isn't necessary. :)

File: git/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+## Git Config
+
+Update your global `.gitconfig` with the content of the `gitconfig.txt`
```
Comment (ghost) [inline git/README.md:11]: The git config file can be used at any configuration level: system, global or local.

A link to git's docs would be nice: https://git-scm.com/docs/git-config

File: README.md [hunk 4]
```diff
@@ -1 +1,21 @@
...
+
+## Contribute
+
+You can contribute too, with your scripts or other nice "tooling for developers", using PR.
+
+Also do not forget to always tests your stuff on your end :P
```
Comment (ghost) [inline README.md]: Have you tested your stuff?

**Comment by ghost** [comment]
Why are the registry files binary?

File: scoop/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+- For already "validated" apps use `scoop.import.all.version.ps1`
+- For all newest version of app `scoop.import.all.newer.ps1` (be aware, it might not been validated with VirusTotal)
+
+> [NOTE]
+> You can specify a folder. If not it will try to import app using the file in the current `backup` folder (e.g.: `.\backup\`)
+
+### Exporting your Apps
+
+If you need to export your app list for a new team member or to use it throughout multiple computer, you only need to use execute the script `scoop.export.all.ps1`
```
Comment (ghost) [inline scoop/README.md]: "multiple computers" -> plural

File: vscode/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# VSCode Extensions
+
+VSCode scripts helps export and import extensions. It's better for onboarding than for using it on a day to day basis. For a daily usage we recommend turning "on Settings Sync..."
```
Comment (ghost) [inline vscode/README.md]: turning "on Settings Sync..." -> turning on "Settings Sync..."

File: vscode/vscode.import.extensions.ps1 [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    )
+
+function Import-Extensions {
+    param(
+        [Parameter(Mandatory = $True)]
+        [String] $Path
+    )
+
+    # import Extensions for VSCode
+    Get-Content "$Path\backup\vscode.extensions.txt" | ForEach-Object { code --install-extension $_ }
```
Comment (ghost) [inline vscode/vscode.import.extensions.ps1]: VSCode must exist in `PATH` for this to work properly.

File: vscode/vscode.export.extensions.ps1 [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    )
+
+function Export-Extensions {
+    param(
+        [Parameter(Mandatory = $True)]
+        [String] $Path
+    )
+
+    # export Extensions for VSCode
+    code --list-extensions | ForEach-Object { "$_" } > "$Path\backup\vscode.extensions.txt"
```
Comment (ghost) [inline vscode/vscode.export.extensions.ps1]: VSCode must exist in `PATH` for this to work properly.

**Comment by ghost** [comment]
Missing VSCode Insiders export script?

File: scoop/scoop.export.all.ps1 [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    param(
+        [Parameter(Mandatory = $True)]
+        [String] $Path
+    )
+
+    # export BUCKETS
+    scoop export | Select-String '\[(\w+)\]' | ForEach-Object { $_.matches.groups[1].value } | Select-Object -unique > "$Path\backup\buckets.txt"
+}
+
+function Export-AppAndVersions {
```
Comment (ghost) [inline scoop/scoop.export.all.ps1]: This function and `Export-AppNameOnly` seem to have their names swapped.

File: README.md (hunk 1 above)
Comment (alexchapar) [reply README.md]: I updated with your suggestions and changed some other stuff in the getting started. I also did some little changes to the blockquote

**Comment by alexchapar** [comment]
> Missing VSCode Insiders export script?

I did change the way scripts for the extensions' export/import for VSCode* ia done... It now includes both. It's kinda "rigid" for now but I think it could be a good start

**Comment by alexchapar** [comment]
@Bibz87-CG, I fixed all your comments, added some stuff and did some testing on my end. It all seems to be good... for now 😄

**Review by ghost — CHANGES_REQUESTED** [review]
> @Bibz87-CG, I fixed all your comments, added some stuff and did some testing on my end. It all seems to be good... for now 😄

I'm still seeing unresolved comments. 😛

File: README.md (hunk 4 above)
Comment (alexchapar) [reply README.md]: As mentionned previously... 
> ... I [...] did some testing on my end. It all seems to be good... for now 😄

Comment (alexchapar) [reply README.md]: unless you found something during your tests ?

File: vscode/vscode.import.extensions.ps1 (hunk 7 above)
Comment (alexchapar) [reply vscode/vscode.import.extensions.ps1]: hmmm... would you suggest a note to add code in `PATH` ? cause trying to find out where the "code" is installed could be tough if not executed within it... ;)

**Comment by alexchapar** [comment]
> Why are the registry files binary?

I don't know... since we can view the file... it seems to not be shown in the `files changed` but we can view them (using the dots)

**Comment by alexchapar** [comment]
> > @Bibz87-CG, I fixed all your comments, added some stuff and did some testing on my end. It all seems to be good... for now 😄
> 
> I'm still seeing unresolved comments. 😛

I did some changes... should be good now... 😄

File: git/README.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# Configuration for Git
+
+[[_TOC_]]
```
Comment (ghost) [inline git/README.md]: Looks like GitHub doesn't support table of content: https://github.com/isaacs/github/issues/215

File: scoop/README.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
+# SCOOP
+
+[[_TOC_]]
```
Comment (ghost) [inline scoop/README.md]: Same here

File: vscode/README.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
+# VSCode Extensions
+
+VSCode scripts helps export and import extensions. It's better for onboarding than for using it on a day to day basis. For a daily usage we recommend turning on "Settings Sync..."
+
+## How to use it
+
+> [!WARNING]
```
Comment (ghost) [inline vscode/README.md]: You might want to replace [!WARNING] and [!NOTE] with proper emojis instead.

Comment (alexchapar) [reply vscode/README.md]: This ia an excellent idea ;)

Comment (alexchapar) [reply vscode/README.md]: It was an excellent idea
I used the emojis based on that suggestion : https://gist.github.com/rxaviers/7360908#gistcomment-3398183

File: git/README.md (hunk 10 above)
Comment (alexchapar) [reply git/README.md]: I removed the [[_TOC_]] since our readme are still little... ;)

File: scoop/README.md (hunk 11 above)
Comment (alexchapar) [reply scoop/README.md]: I removed the [[_TOC_]] since our readme are still little... ;)

**Review by ghost — APPROVED** [review]
All good! :)

**Pull request merged**